### PR TITLE
Issue #15061: Add new input for pattern variables on AnnotationOnSame…

### LINKED ISSUE #15061: Add Check Support for Java 21 Unnamed Variables & Patterns Syntax: AnnotationOnSameLineCheck
child of https://github.com/checkstyle/checkstyle/issues/14942 
Check documentation: https://checkstyle.org/checks/annotation/annotationonsameline.html#AnnotationOnSameLine

```
PS D:\CS\test> javac src/Test.java
PS D:\CS\test> cat config.xml                                                   
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <property name="charset" value="UTF-8"/>
    <module name="TreeWalker">
             <module name="AnnotationOnSameLine">
        </module>
    </module>
</module>
PS D:\CS\test> cat src/Test.java  
record ColoredPoint(int p, int x, String c) { }
record Rectangle(ColoredPoint upperLeft, ColoredPoint lowerRight) { }
public class Test {
    void test(Object obj) {
        @Deprecated
        int _ = 0;

        if (obj instanceof ColoredPoint(@Deprecated // expected violation
                                          int _, _,_)){
        }
    }
}
PS D:\CS\test> java  -jar checkstyle-10.17.0-all.jar -c config.xml src/Test.java
Starting audit...
[ERROR] D:\CS\test\src\Test.java:5:9: Annotation 'Deprecated' should be on the same line with its target. [AnnotationOnSameLine]
Audit done.
Checkstyle ends with 1 errors.
PS D:\CS\test> 

```
---

This is not very specific to unnamed patterns. but for patterns in general.
I expect a violation on the annotations on the pattern variable. The annotation should be on the same line as the pattern. We should add `PATTERN_VARIABLE_DEF` in the tokens

### PR DESCRIPTION
**Pull request by mahfouz72**
closes #15061

### CODE REVIEW

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/annotation/annotationonsameline/InputAnnotationOnSameLinePatternVariables.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+tokens = (default)CLASS_DEF, INTERFACE_DEF, ENUM_DEF, METHOD_DEF, CTOR_DEF, \
+         VARIABLE_DEF, RECORD_DEF, COMPACT_CTOR_DEF
+
+
+*/
+
+//non-compiled with javac: Compilable with Java21
+package com.puppycrawl.tools.checkstyle.checks.annotation.annotationonsameline;
+
+public class InputAnnotationOnSameLinePatternVariables {
```
Comment (nrmancuso) [inline src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/annotation/annotationonsameline/InputAnnotationOnSameLinePatternVariables.java:12]: Let's add some unnamed variables in here too: `@Deprecated int _ = 0;` as well as unnamed record components.

**Review by nrmancuso — CHANGES_REQUESTED** [review]
Item:

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/annotation/annotationonsameline/InputAnnotationOnSameLinePatternVariables.java (hunk 1 above)
Comment (mahfouz72) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/annotation/annotationonsameline/InputAnnotationOnSameLinePatternVariables.java:12]: done

**Comment by mahfouz72** [comment]
semaphoreci has stopped

**Review by nrmancuso — APPROVED** [review]

**Pull request merged**